libraries/core_libs/network/include/network/tarcap/taraxa_capability.hpp

```diff
@@ -109,6 +109,8 @@ class TaraxaCapability : public dev::p2p::CapabilityFace {
                               const std::shared_ptr<DagBlockManager> &dag_blk_mgr,
                               const std::shared_ptr<TransactionManager> &trx_mgr, addr_t const &node_addr);
 
+  inline bool filterSyncIrrelevantPackets(SubprotocolPacketType packet_type);
```

Comment:
Why would you create inline class method ? I saw it in multiple PR's of yours... Inline can improve speed/size/etc... but it can also make it much worse. Compiler can actually make any function inline so why not let the decision to the compiler optimization ?

Reply:
inline is just suggestion for compiler https://en.cppreference.com/w/cpp/language/inline 
```
Since this meaning of the keyword inline is non-binding, compilers are free to use inline substitution for any function that's not marked inline, and are free to generate function calls to any function marked inline. Those optimization choices do not change the rules regarding multiple definitions and shared statics listed above. 
```

Reply:
because it is just more complicated if and it is in function just to have more clear code and call this function is much more expensive then doing that switch

Reply:
I pretty much agree with [this answer](https://stackoverflow.com/questions/1759300/when-should-i-write-the-keyword-inline-for-a-function-method) (1k+ upvotes), which says:

It is said that inline hints to the compiler that you think the function should be inlined. That may have been true in 1998, but a decade later the compiler needs no such hints. Not to mention humans are usually wrong when it comes to optimizing code, so most compilers flat out ignore the 'hint'.

- When should I write the keyword 'inline' for a function/method in C++?

Only when you want the function to be defined in a header. More exactly only when the function's definition can show up in multiple translation units. It's a good idea to define small (as in one liner) functions in the header file as it gives the compiler more information to work with while optimizing your code. It also increases compilation time.

- When should I not write the keyword 'inline' for a function/method in C++?

Don't add inline just because you think your code will run faster if the compiler inlines it.

- When will the compiler not know when to make a function/method 'inline'?

Generally, the compiler will be able to do this better than you. However, the compiler doesn't have the option to inline code if it doesn't have the function definition. In maximally optimized code usually all private methods are inlined whether you ask for it or not.

but I am willing to change my opinion if you provde some better source of explanation and arguments :)

Reply:
This [microsoft doc](https://docs.microsoft.com/en-us/cpp/cpp/inline-functions-cpp?view=msvc-160) on the other hand claims it is worth it. Anyway I dont know that much about it, so lets use it if you think it is worth it too :) But when googling it, I only found a form where you use inline keyword only in cpp file, not inside header declaration...